dcos-log/api/v2/handlers.go

```diff
@@ -2,6 +2,7 @@ package v2
 
 import (
 	"context"
+	"errors"
```

Comment:
Just as a general suggestion -- this is a pretty big commit that touches a number of things. I think in the future this could be broken up into smaller commits according to scope.